src/openms/include/OpenMS/FORMAT/FileHandler.h

```diff
@@ -166,76 +205,222 @@ namespace OpenMS
 
       @param filename The name of the file to store the data in.
       @param exp The experiment to store.
+      @param allowed_types a FileTypeList containing the types of files to support. If not empty the extension, and failing that the contents are checked to make sure they match
       @param log Progress logging mode
 
       @exception Exception::UnableToCreateFile is thrown if the file could not be written
     */
-    void storeExperiment(const String& filename, const MSExperiment& exp, ProgressLogger::LogType log = ProgressLogger::NONE);
+    bool storeExperiment(const String& filename, const PeakMap& exp, const std::vector<FileTypes::Type> allowed_types = std::vector<FileTypes::Type>(), ProgressLogger::LogType log = ProgressLogger::NONE);
 
     /**
       @brief Loads a file into a FeatureMap
 
       @param filename the file name of the file to load.
       @param map The FeatureMap to load the data into.
-      @param force_type Forces to load the file with that file type. If no type is forced, it is determined from the extension (or from the content if that fails).
 
       @return true if the file could be loaded, false otherwise
 
       @exception Exception::FileNotFound is thrown if the file could not be opened
       @exception Exception::ParseError is thrown if an error occurs during parsing
     */
-    bool loadFeatures(const String& filename, FeatureMap& map, FileTypes::Type force_type = FileTypes::UNKNOWN);
+    bool loadFeatures(const String& filename, FeatureMap& map, const std::vector<FileTypes::Type> allowed_types = std::vector<FileTypes::Type>(), ProgressLogger::LogType log = ProgressLogger::NONE);
 
     /**
       @brief Store a FeatureMap
 
       @param filename the file name of the file to write.
       @param map The FeatureMap to store.
+      @param allowed_types a FileTypeList containing the types of files to support. If not empty the extension, and failing that the contents are checked to make sure they match
+      @param log Progress logging mode
+
 
       @return true if the file could be stored, false otherwise
 
       @exception Exception::UnableToCreateFile is thrown if the file could not be written
     */
-    bool storeFeatures(const String& filename, const FeatureMap& map);
+    bool storeFeatures(const String& filename, const FeatureMap& map, const std::vector<FileTypes::Type> allowed_types = std::vector<FileTypes::Type>(), ProgressLogger::LogType log = ProgressLogger::NONE);
+
+    /**
+      @brief Loads a file into a ConsensusMap
+
+      @param filename the file name of the file to load.
+      @param map The ConsensMap to load the data into.
+      @param allowed_types a FileTypeList containing the types of files to support. If not empty the extension, and failing that the contents are checked to make sure they match
```

Comment:
I don't understand this comment :)

Reply:
Does this look better:
 a vector<FileTypes::Files> containing the types supported where this function is called. If the vector is empty, load from any type that we have a handler for. Otherwise @getType() is called internally to check the type

Reply:
```
allowed_types list used to restrict the set of accepted file types during loading. If empty, no restrictions are imposed and files are loaded according to extension or detected by file content.
```

Reply:
maybe something like this?